jetstream/README.md

```diff
@@ -603,6 +609,193 @@ ackF, err = js.PublishAsync("ORDERS.new", []byte("hello"))
+delete markers.
+- `UpdatesOnly` will have the key watcher only pass updates on values
+(without latest values when started).
+- `MetaOnly` will have the key watcher retrieve only the entry meta
+data, not the entry value.
+- `ResumeFromRevision` instructs the key watcher to resume from a
+specific revision number.
+
+```go
+js, _ := jetstream.New(nc)
+ctx := context.Background()
+kv, _ := js.CreateKeyValue(ctx, jetstream.KeyValueConfig{Bucket: "profiles"})
+
+kv.Put(ctx, "sue.color", []byte("blue"))
+
+// A watcher can be created to watch for changes on a given key or the whole bucket
+// Watcher will receive a notification on a channel when a change occurs
+// By default, watcher will return initial values for all matching keys.
+// Watcher can be configured to only return updates by using jetstream.UpdatesOnly() option.
+watcher, _ := kv.Watch(ctx, "sue.*")
+defer watcher.Stop()
+
+kv.Put(ctx, "sue.age", []byte("43"))
+kv.Put(ctx, "sue.color", []byte("red"))
+
+// First, the watcher sends initial values for all matching keys
+// In this case, it will send a single entry for `sue.color`.
+entry := <-watcher.Updates()
+// Prints `sue.color @ 1 -> "blue"`
+fmt.Printf("%s @ %d -> %q\n", entry.Key(), entry.Revision(), string(entry.Value()))
+
+// After all initial values have been sent, watcher will send nil on the channel.
+entry = <-watcher.Updates()
+if entry != nil {
+    fmt.Println("Unexpected entry received")
+}
+
+// After that, watcher will send updates when changes occur
+// In this case, it will send an entry for `sue.color` and `sue.age`.
+
+entry = <-watcher.Updates()
+// Prints `sue.age @ 2 -> "43"`
+fmt.Printf("%s @ %d -> %q\n", entry.Key(), entry.Revision(), string(entry.Value()))
+
+entry = <-watcher.Updates()
+// Prints `sue.color @ 3 -> "red"`
+fmt.Printf("%s @ %d -> %q\n", entry.Key(), entry.Revision(), string(entry.Value()))
+```
+
+### Additional operations on a bucket
+
+In addition to basic CRUD operations and watching for changes, KV buckets
+support several additional operations:
+
+- `ListKeys` will return all keys in a bucket"
+
+```go
+js, _ := jetstream.New(nc)
+ctx := context.Background()
+kv, _ := js.CreateKeyValue(ctx, jetstream.KeyValueConfig{Bucket: "profiles"})
+
+kv.Put(ctx, "sue.color", []byte("blue"))
+kv.Put(ctx, "sue.age", []byte("43"))
+kv.Put(ctx, "bucket", []byte("profiles"))
+
+keys, _ := kv.ListKeys(ctx)
+
+// Prints all 3 keys
+for key := range keys.Keys() {
+    fmt.Println(key)
+}
+```
+
+- `Purge` and `PurgeDeletes` for removing all keys from a bucket
+
+```go
+js, _ := jetstream.New(nc)
+ctx := context.Background()
+kv, _ := js.CreateKeyValue(ctx, jetstream.KeyValueConfig{Bucket: "profiles"})
+
+kv.Put(ctx, "sue.color", []byte("blue"))
+kv.Put(ctx, "sue.age", []byte("43"))
+kv.Put(ctx, "bucket", []byte("profiles"))
+
+// Purge will remove all keys from a bucket
+// The latest revision of each key will be kept
+// with a delete marker, all previous revisions will be removed
```

Comment:
I would make a distinction between purge and delete clear in a sense that one is destructive, other is not.

Reply:
I will add further clarification in method descriptions, only slight refinement here. I wouldn't like to add too much commentary in code examples.